precompile/allow_list.go

```diff
@@ -83,7 +84,7 @@ func (s AllowListRole) IsAdmin() bool {
 // IsEnabled returns true if [s] indicates that it has permission to access the resource.
 func (s AllowListRole) IsEnabled() bool {
 	switch s {
-	case AllowListEnabled:
+	case AllowListAdmin, AllowListEnabled:
```

Comment:
I feel like this is kind-of misleading since both of role name and function contains `Enabled` however the function returns true for `Admin` as well. Plus do we really want admins to be able to mint tokens?

Reply:
Admin is a strictly higher and should be considered enabled as well. If we don't consider it enabled, then the UX for an admin will be that they have to create a new address, mark it as enabled, and then mint funds from that address. It's much simpler if we just treat Admin as a strictly higher privilege level.